python/cudf/cudf/core/column/lists.py

```diff
@@ -453,18 +454,73 @@ def contains(self, search_key: ScalarLike) -> ParentType:
             raise
         return res
 
-    def index(self, search_key: ScalarLike) -> ParentType:
-        search_key = cudf.Scalar(search_key)
+    def index(self, search_key: Union[ScalarLike, ColumnLike]) -> ParentType:
+        """
+        Returns integers representing the index of the search key for each row.
+
+        If ``search_key`` is a sequence, it must be the same length as the
+        Series and ``search_key[i]`` represents the search key for the
+        ``i``-th row of the Series.
+
+        If the search key is not contained in a row, -1 is returned. If either
+        the row or the search key are null, <NA> is returned. If the search key
+        is contained multiple times, the smallest matching index is returned.
+
+        Parameters
+        ----------
+        search_key : scalar or sequence of scalars
+            Element or elements being searched for in each row of the list
+            column
+
+        Returns
+        -------
+        Series or Index
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], [3, 4, 5], [4, 5, 6]])
+        >>> s.list.index(4)
+        0   -1
+        1    1
+        2    0
+        dtype: int32
+
+        >>> s = cudf.Series([["a", "b", "c"], ["x", "y", "z"]])
+        >>> s.list.index(["b", "z"])
+        0    1
+        1    2
+        dtype: int32
+
+        >>> s = cudf.Series([[4, 5, 6], None, [-3, -2, -1]])
+        >>> s.list.index([None, 3, -2])
+        0    <NA>
+        1    <NA>
+        2       1
+        dtype: int32
+
+        Notes
+        -----
+        ``index`` only supports list search operations on numeric types,
+        decimals, chrono types, and strings.
```

Comment:
"Chrono" is a C++ / libcudf term and doesn't appear in the cuDF Python docs. The Python docs discuss [datetimes and timedeltas](https://docs.rapids.ai/api/cudf/nightly/basics/basics.html).

To clarify my own understanding, what types are _not_ supported here? It looks like this is exhaustive of the scalar types we support unless there's some catch for categorical or bool?
```suggestion
        decimals, datetimes, timedeltas, and strings.
```

Reply:
Yep, upon further investigation it looks like bools are supported. But when using a list scalar as a search key on a multi-level nested list series `index_of` throws:
```List search operations are only supported on numeric types, decimals, chrono types, and strings.``` 

Do you know how I might go about testing categorical types?